File: 0102 binary-tree-level-order-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans = new ArrayList<>();
        if(root == null) return ans;
        Queue<TreeNode> que = new LinkedList<>();
        que.offer(root);
        while(!que.isEmpty()) {
            int levelSize = que.size();
            List ians = new ArrayList<>();
            for(int i=0;i<levelSize;i++) {
                TreeNode curr = que.poll();
                if(curr.left!=null) que.offer(curr.left);
                if(curr.right!=null) que.offer(curr.right);
                ians.add(curr.val);
            }
            ans.add(ians);
        }
        return ans;
    }
}

File: 0102 binary-tree-level-order-traversal/solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        IList<IList<int>> ans = new List<IList<int>>();
        if (root == null) return ans;
        LinkedList<TreeNode> queue = new LinkedList<TreeNode>();
        queue.AddLast(root);
        while(queue.Count > 0) {
            int qsize = queue.Count;
            IList<int> ians = new List<int>();
            for(int i=0;i<qsize;i++) {
                TreeNode curr = queue.First.Value;
                queue.RemoveFirst();
                if(curr.left!=null) queue.AddLast(curr.left);
                if(curr.right!=null)  queue.AddLast(curr.right);
                ians.Add(curr.val);
            }
            ans.Add(ians);
        }
        return ans;
    }
}
